programs/server/config.xml

```diff
@@ -66,69 +66,289 @@
+              -->
+            <!--
+            <allow_proxies>
+                <my_load_balancer />
+                <my_http_proxy />
+            </allow_proxies>
+            -->
+        </plain_native_tcp>
+
+        <!-- Interface for interactions using secure native TCP protocol.
+             You have to configure certificate to enable this interface.
+             See the openSSL section below.
+             See also 'plain_native_tcp' for insecure connections.
+             All parameters relevant to a TCP-based protocol can be specified here. See 'plain_native_tcp'.
+          -->
+        <!--
+        <secure_native_tcp>
+            <protocol>native_tcp</protocol>
+            <port>9440</port>
+            <enable_tls>yes</enable_tls>
+        </secure_native_tcp>
+        -->
+
+        <!-- Interface for interactions using plain native HTTP API.
+             This interface is also used by ODBC and JDBC drivers (DataGrip, Dbeaver, ...)
+             and by most of web interfaces (embedded UI, Grafana, Redash, ...).
+             See also 'secure_native_http' for secure connections.
+             All parameters relevant to a TCP-based protocol can be specified here. See 'plain_native_tcp'.
+          -->
+        <plain_native_http>
+            <protocol>native_http</protocol>
+            <port>8123</port>
+            <enable_tls>no</enable_tls>
+        </plain_native_http>
+
+        <!-- Interface for interactions using secure native HTTPS API.
+             You have to configure certificate to enable this interface.
+             See the openSSL section below.
+             See also 'plain_native_http' for insecure connections.
+             All parameters relevant to a HTTP-based protocol can be specified here. See 'plain_native_http'.
+          -->
+        <!--
+        <secure_native_http>
+            <protocol>native_http</protocol>
+            <port>8443</port>
+            <enable_tls>yes</enable_tls>
+        </secure_native_http>
+        -->
+
+        <!-- Interface for interactions using plain (or secure, if 'grpc.enable_ssl' is set to 'true') native gRPC protocol.
+             See also 'grpc' for further configutarion.
+             Some parameters relevant to a TCP-based protocol ('listen', 'port', 'listen_try') can be specified here. See 'plain_native_tcp'.
+          -->
+        <!--
+        <plain_native_grpc>
+            <protocol>native_grpc</protocol>
+            <port>9100</port>
+        </plain_native_grpc>
+        -->
+
+        <!-- Interface for interactions using MySQL compatibility protocol.
+             ClickHouse will pretend to be MySQL for applications connecting to this port.
+             All parameters relevant to a TCP-based protocol can be specified here. See 'plain_native_tcp'.
+          -->
+        <plain_mysql>
+            <protocol>mysql</protocol>
+            <port>9004</port>
+            <enable_tls>no</enable_tls> <!-- TLS connections are not supported. -->
```

Comment:
https://dev.mysql.com/doc/refman/8.0/en/using-encrypted-connections.html

Reply:
The current implementation of MySQL protocol in ClickHouse seemingly doesn't support secure connections. Adding it is out of the scope of this PR.